packages/ui/src/pages/Beans/BeansPage.tsx

```diff
@@ -41,7 +41,7 @@ export const BeansPage: FunctionComponent = () => {
         entity.parent.beans = model;
       } else {
         const entity = findBeansEntity();
-        entity?.parent.beans.length === 0 && beansAwareResource.deleteBeansEntity(entity);
+        entity && beansAwareResource.deleteBeansEntity(entity);
```

Comment:
Wouldn't it be easier to handle this in the `deleteBeansEntity` method itself? so we don't need to evaluate whether the entity is defined or not.

Reply:
Maybe wait and see, we might want to handle multiple `beans` entry in a better UI, not yet sure what's the best for the boundary there.